js/statements/createmacro.js

```diff
@@ -0,0 +1,48 @@
+
+function GenerateCreateMacro(options = {}) {
+	return Diagram([
+		AutomaticStack([
+			Keyword("CREATE"),
+			Choice(0, [
+				Keyword("MACRO"),
+				Keyword("FUNCTION"),
+				]
+			),
+			Optional(Sequence([
+				Expression("schema-name"),
+				Keyword(".")
+			]), "skip"),
+			Expression("macro-name"),
+			Keyword("("),
+			Optional(Sequence([
+				OneOrMore(Sequence([
```

Comment:
I think this should be ZeroOrMore, rather than OneOrMore, no?

Reply:
Didn't know about ZeroOrMore! I guess Optional + OneOrMore is a bit hacky. Done.